File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/kafka/producers/model/Adresse.java
package no.nav.melosys.eessi.kafka.producers.model;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Adresse {
    private String by;
    private String bygning;
    private String gate;
    private String land;
    private String postnummer;
    private String region;
    private String type;

    public Adresse(no.nav.melosys.eessi.models.sed.nav.Adresse adresse) {
        this(adresse.getBy(), adresse.getBygning(), adresse.getGate(), adresse.getLand(),
            adresse.getPostnummer(), adresse.getRegion(), adresse.getType());
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/kafka/producers/model/Adresse.kt
package no.nav.melosys.eessi.kafka.producers.model

data class Adresse(
    var by: String?,
    var bygning: String?,
    var gate: String?,
    var land: String?,
    var postnummer: String?,
    var region: String?,
    var type: String?
) {
    constructor(adresse: no.nav.melosys.eessi.models.sed.nav.Adresse) : this(
        adresse.by,
        adresse.bygning,
        adresse.gate,
        adresse.land,
        adresse.postnummer,
        adresse.region,
        adresse.type
    )
}
